### PR TITLE
Added ParentWIndow to IsBrokeravailable

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
Fixes # 3320

**Changes proposed in this request**
Added parent window parameter to the API

**Testing**
Manual testing using IntuneMAM iOS App

**Performance impact**
none

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/UI/CoreUIParent.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 namespace Microsoft.Identity.Client.UI
 {
-    internal class CoreUIParent
+    public class CoreUIParent
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/UI/CoreUIParent.cs:20]: please don't make this class public

File: src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs [hunk 2]
```diff
@@ -339,10 +341,10 @@ public PublicClientApplicationBuilder WithKerberosTicketClaim(string servicePrin
         /// <item><description>In mobile apps, the device must be Intune joined and Authenticator or Company Portal must be installed. See https://aka.ms/msal-brokers </description></item>
         /// </list>
         /// </remarks>
-        public bool IsBrokerAvailable()
+        public bool IsBrokerAvailable(CoreUIParent parentWindow = null)
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:344]: No, please don't make this public.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs:344]: Maybe we need a static helper `PCA.IsBrokerAvailable(authority, parent)` ?

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
do not make coreuiparent public

**Comment by SameerK-MSFT** [comment]
This is under discussion. Converting to draft

**Comment by bgavrilMS** [comment]
@SameerK-MSFT - please fix this PR or close it.

**Pull request closed without merging**